pac4j-oidc/src/main/java/org/pac4j/oidc/config/OidcConfiguration.java

```diff
@@ -282,11 +299,16 @@ public void setResourceRetriever(final ResourceRetriever resourceRetriever) {
     }
 
     public String getResponseType() {
-        return responseType;
+        return responseType.toString();
     }
 
     public void setResponseType(final String responseType) {
-        this.responseType = responseType;
+        responseTypeException = null;
+        try {
+            this.responseType = ResponseType.parse(responseType);
+        } catch (ParseException e) {
+            responseTypeException = new TechnicalException("Unrecognised responseType: " + responseType, e);
```

Comment:
Let's make things more simple: directly throw the `TechnicalException` here.

Reply:
This changes the behaviour of the code to throw exceptions in the setters rather than in the init method like now, however if you're happy the change in behaviour is not a problem I'll update it.

Reply:
yes, but in any case, the exception will be thrown when the object is initialized so the behavior is changed. So I'd prefer to throw it as early as possible. Given the fact that this exception should never be thrown.